distributed/scheduler.py

```diff
@@ -7477,7 +7504,33 @@ def decide_worker(
         for ws in candidates:
             break
     else:
+        group: TaskGroup = ts._group
+        ws = group._last_worker
+
+        total_nthreads = sum(
+            wws._nthreads for wws in candidates
+        )  # TODO get `self._total_threads` from scheduler? Though that doesn't account for worker restrictions.
+
+        group_tasks_per_worker = len(group) / total_nthreads
+
+        # Try to schedule sibling root-like tasks on the same workers, so subsequent reduction tasks
+        # don't require data transfer. Assumes `decide_worker` is being called in priority order.
+        if (
+            ws is not None  # there is a previous worker
+            and group._last_worker_tasks_left > 0  # previous worker not fully assigned
+            and ts._dependents  # task has dependents
+            and group_tasks_per_worker > 1  # group is larger than cluster
+            and (  # is a root-like task (task group depends on very few tasks)
+                sum(map(len, group._dependencies)) < 5  # TODO what number
+            )
+        ):
+            group._last_worker_tasks_left -= 1
```

Comment:
In general I think that having a counter like this opens us up to failures in some complex situations. 

For example, what happens when we lose a few workers and need to recompute these tasks and this count goes negative?

I would encourage us to keep thinking about comparing occupancy between the best and recent workers.  That feels like a strictly local decision that might be more robust.

Reply:
If the problem is that duration_average is unknown that's fine, we should replace it with whatever is used to increment the worker's occupancy when the previous tasks were added.

Reply:
Yeah, this will probably get thrown off by both new workers arriving and workers leaving. I'll think more about how to use something less stateful.